src/main/java/net/dv8tion/jda/api/entities/MessageChannel.java

```diff
@@ -643,9 +643,13 @@ default MessageAction sendFile(@Nonnull File file, @Nonnull String fileName, @No
     {
         Checks.notNull(file, "file");
         Checks.check(file.exists() && file.canRead(),
-            "Provided file is either null, doesn't exist or is not readable!");
-        Checks.check(file.length() <= getJDA().getSelfUser().getAllowedFileSize(),
-            "File is too big! Max file-size is 8 MiB for normal and 50 MiB for nitro users");
+            "Provided file doesn't exist or cannot be read!");
+        long allowedSelfUserFileSize = getJDA().getSelfUser().getAllowedFileSize();
+        long allowedFileSize = this instanceof GuildChannel
+                ? Math.max(allowedSelfUserFileSize, ((GuildChannel) this).getGuild().getMaxFileSize())
+                : allowedSelfUserFileSize;
```

Comment:
Instead of doing this here, you should override this method in `AbstractChannelImpl` and `PrivateChannelImpl` and do the checks there.

Reply:
@MinnDevelopment are you sure? If i wanted to override the method in both classes, they would need to implement MessageChannel which they don't.

Reply:
I assume he means get rid of the ternary check and instead have:
- `sendFile` in `TextChannelImpl` which uses the highest file size between the guild one and the self user one
- `sendFile` in `PrivateChannelImpl` which just uses the self user file size limit.

Reply:
Should be good now.